src/cephadm/cephadm

```diff
@@ -3267,7 +3267,7 @@ def _list_ipv6_networks():
 
 def _parse_ipv6_route(routes, ips):
     r = {}  # type: Dict[str,List[str]]
-    route_p = re.compile(r'^(\S+) dev (\S+) proto (\S+) metric (\S+) pref (\S+)$')
+    route_p = re.compile(r'^(\S+) dev (\S+) proto (\S+) metric (\S+) .*pref (\S+)$')
```

Comment:
we could probably greatly simplify the regex and not collect a bunch of groups we never check.. but grabbing them in case we want that info one day.

Reply:
yikes!

The ip command can print JSON output. Could that be used to replace this complex regexp?

```
$ ip -j -6 route ls
[{"dst":"::1","dev":"lo","protocol":"kernel","metric":256,"flags":[],"pref":"medium"},{"dst":"fe80::/64","dev":"br0","protocol":"kernel","metric":256,"flags":[],"pref":"medium"},{"dst":"fe80::/64","dev":"eno1","protocol":"kernel","metric":256,"flags":[],"pref":"medium"},{"dst":"fe80::/64","dev":"cni-podman0","protocol":"kernel","metric":256,"flags":["linkdown"],"pref":"medium"},{"dst":"fe80::/64","dev":"vnet0","protocol":"kernel","metric":256,"flags":[],"pref":"medium"},{"dst":"fe80::/64","dev":"vnet1","protocol":"kernel","metric":256,"flags":[],"pref":"medium"},{"dst":"fe80::/64","dev":"vnet2","protocol":"kernel","metric":256,"flags":[],"pref":"medium"},{"dst":"fe80::/64","dev":"vnet3","protocol":"kernel","metric":256,"flags":[],"pref":"medium"},{"dst":"fe80::/64","dev":"vnet4","protocol":"kernel","metric":256,"flags":[],"pref":"medium"},{"dst":"fe80::/64","dev":"vnet5","protocol":"kernel","metric":256,"flags":[],"pref":"medium"},{"dst":"fe80::/64","dev":"vnet6","protocol":"kernel","metric":256,"flags":[],"pref":"medium"},{"dst":"fe80::/64","dev":"vnet7","protocol":"kernel","metric":256,"flags":[],"pref":"medium"}]
```
```

Reply:
100% agree here, but apparently there is some version of IP on ubuntu (18:04 apparently) [0] that doesn't have the `-j` command, so we're stuck with regex's until we decide not to support it.

The code will be much better when we can use json.

[0] - https://github.com/ceph/ceph/blob/master/src/cephadm/cephadm#L3231-L3232

Reply:
that's unfortunate :(